src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -38,13 +38,13 @@ public function dispatch($message, array $stamps = []): Envelope
         ];
 
         try {
-            return $this->decoratedBus->dispatch($message, $stamps);
+            return $envelope = $this->decoratedBus->dispatch($message, $stamps);
         } catch (\Throwable $e) {
             $context['exception'] = $e;
 
             throw $e;
         } finally {
-            $this->dispatchedMessages[] = $context;
+            $this->dispatchedMessages[] = $context + ['stamps_after_dispatch' => array_values($envelope->all())];
```

Comment:
do we want to remove the stamps that were here before dispatch?

Reply:
I don't think so. The exhaustive information about all stamps present in the envelope might be useful, as middleware can remove/replace those. In any case, it's probably less confusing to show all.